File: backend/src/models/Report.js
const mongoose = require('mongoose');

const ReportSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  type: {
    type: String,
    enum: ['daily', 'weekly', 'monthly'],
    required: true,
  },
  period: {
    type: String,
    required: true,
    // 格式: daily-YYYY-MM-DD, weekly-YYYY-WW, monthly-YYYY-MM
  },
  title: {
    type: String,
    required: true,
  },
  content: {
    type: String,
    required: true,
  },
  polishedContent: {
    type: String,
    default: null,
  },
  tasks: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Task',
  }],
  statistics: {
    totalTasks: {
      type: Number,
      default: 0,
    },
    completedTasks: {
      type: Number,
      default: 0,
    },
    inProgressTasks: {
      type: Number,
      default: 0,
    },
    overdueTasks: {
      type: Number,
      default: 0,
    },
    completionRate: {
      type: Number,
      default: 0,
    },
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
  updatedAt: {
    type: Date,
    default: Date.now,
  },
});

// 更新报告时自动更新updatedAt字段
ReportSchema.pre('save', function (next) {
  if (this.isModified()) {
    this.updatedAt = Date.now();
  }
  next();
});

module.exports = mongoose.model('Report', ReportSchema);

File: backend-go/src/models/Report.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type ReportType string

const (
	DailyReport   ReportType = "daily"
	WeeklyReport  ReportType = "weekly"
	MonthlyReport ReportType = "monthly"
)

type ReportStatistics struct {
	TotalTasks       int `json:"totalTasks" bson:"totalTasks"`
	CompletedTasks   int `json:"completedTasks" bson:"completedTasks"`
	InProgressTasks  int `json:"inProgressTasks" bson:"inProgressTasks"`
	OverdueTasks     int `json:"overdueTasks" bson:"overdueTasks"`
	CompletionRate   int `json:"completionRate" bson:"completionRate"`
}

type Report struct {
	ID              primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	UserID          primitive.ObjectID `json:"userId" bson:"userId" validate:"required"`
	Type            ReportType         `json:"type" bson:"type" validate:"required"`
	Period          string             `json:"period" bson:"period" validate:"required"`
	Title           string             `json:"title" bson:"title" validate:"required"`
	Content         string             `json:"content" bson:"content" validate:"required"`
	PolishedContent *string            `json:"polishedContent" bson:"polishedContent"`
	Tasks           []primitive.ObjectID `json:"tasks" bson:"tasks"`
	Statistics      ReportStatistics   `json:"statistics" bson:"statistics"`
	CreatedAt       time.Time          `json:"createdAt" bson:"createdAt"`
	UpdatedAt       time.Time          `json:"updatedAt" bson:"updatedAt"`
}